WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -28,12 +29,10 @@ public Round(RoundParameters roundParameters)
 			VsizeCredentialIssuer = new(new(RoundParameters.Random), RoundParameters.Random, MaxRegistrableVsize);
 			AmountCredentialIssuerParameters = AmountCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
 			VsizeCredentialIssuerParameters = VsizeCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
-
-			Id = CalculateHash();
 		}
 
+		public uint256 Id => _id ??= CalculateHash();
 		public MultipartyTransactionState CoinjoinState { get; set; }
-		public uint256 Id { get; }
```

Comment:
I don't understand what this change to how `Id` is computed is doing, is it a leftover from a refactoring? Seems to me like it stay using the assignment in the constructor instead?

Reply:
This is my question too, this makes no sense to me compared to calculate it in the constructor.

Reply:
If I let it as it was then it computes a different hash but at this moment i don't remember why.